influxdb_iox/tests/query_tests2/setups.rs

```diff
@@ -931,5 +935,96 @@ pub static SETUPS: Lazy<HashMap<SetupName, SetupSteps>> = Lazy::new(|| {
                 },
             ],
         ),
+        (
+            "ThreeChunksWithRetention",
+            vec![
+                Step::RecordNumParquetFiles,
+                Step::WriteLineProtocol(RETENTION_SETUP.lp_partially_inside.clone()),
+                Step::Persist,
+                Step::WaitForPersisted2 {
+                    expected_increase: 1,
+                },
+                Step::RecordNumParquetFiles,
+                Step::WriteLineProtocol(RETENTION_SETUP.lp_fully_inside.clone()),
+                Step::Persist,
+                Step::WaitForPersisted2 {
+                    expected_increase: 1,
+                },
+                Step::RecordNumParquetFiles,
+                Step::WriteLineProtocol(RETENTION_SETUP.lp_fully_outside.clone()),
+                Step::Persist,
+                Step::WaitForPersisted2 {
+                    expected_increase: 1,
+                },
+                Step::SetRetention(Some(RETENTION_SETUP.retention_period_ns)),
+            ],
+        ),
     ])
 });
+
+/// Holds parameters for retention period. Tests based on this need to
+/// be run within the hour of being created.
+///
+/// ```text                  (cut off)                    (now)
+/// time: -----------------------|--------------------------|>
+///
+/// partially_inside:   |-----------------|
+///
+/// fully_inside:                 +1h |-----------|
+///
+/// fully_outside: |---------|
+/// ```
+struct RetentionSetup {
+    /// the retention period, relative to now() that the three data
+    /// chunks fall inside/outside
+    retention_period_ns: i64,
+
+    /// lineprotocol data partially inside retention
+    lp_partially_inside: String,
+
+    /// lineprotocol data fully inside (included in query)
+    lp_fully_inside: String,
+
+    /// lineprotocol data fully outside (excluded from query)
+    lp_fully_outside: String,
+}
+
+impl RetentionSetup {
+    fn new() -> Self {
+        let retention_period_1_hour_ns = 3600 * 1_000_000_000;
+
+        // Data is relative to this particular time stamp
+        let cutoff = Time::from_rfc3339("2022-01-01T00:00:00+00:00")
+            .unwrap()
+            .timestamp_nanos();
+        let inside_retention = cutoff + retention_period_1_hour_ns; // 1 hour before it is outside retention
+        let outside_retention = cutoff - 10; // before retention
+
+        let lp_partially_inside = format!(
+            "cpu,host=a load=1 {inside_retention}\n\
+             cpu,host=aa load=11 {outside_retention}"
+        );
+
+        let lp_fully_inside = format!(
+            "cpu,host=b load=2 {inside_retention}\n\
+             cpu,host=bb load=21 {inside_retention}"
+        );
+
+        let lp_fully_outside = format!(
+            "cpu,host=z load=3 {outside_retention}\n\
+             cpu,host=zz load=31 {outside_retention}"
+        );
+
+        // Set retention period to be the cutoff (though as time
+        // advanges that will be earlier and earlier
```

Comment:
I think "advanges" is supposed to be "advances", the closing paren is missing, and I'm again struggling with "earlier"... and the comment on line 1023 seems to be more applicable to this line? I think this'd be better as something like:

```suggestion
        // Set retention period to be the cutoff, relative to now. As real-life time
        // advances, the retention period used in this test will get larger.
```

Reply:
Yeah that was akward -- I reworded to

```rust
        // Set retention period to be at the cutoff date. Note that
        // since real world time advances, after 1 hour of real world
        // time the data that is inside the retention interval will
        // move outside (and thus not appear in queries).
        //
        // Thus this setup is only valid for 1 hour.
```

Which I think makes it clearer what is going on